geocruncher/ComputeIntersections.py

```diff
@@ -54,19 +54,19 @@ def computeRankMatrix(index):
         # impact of setup code above is insignificant, so we don't profile it in any specific step
 
         if not isOnYAxis:
+            slope = (yCoord[0] - yCoord[1]) / (xCoord[0] - xCoord[1])
+            xSliceRange = np.linspace(xCoord[0], xCoord[1], nPoints)
+            z, x = np.meshgrid(zSliceRange, xSliceRange)
```

Comment:
Nice catch. I originally moved them in the if because it seemed more efficient, but then reverted my change.
This is because the x, y and z variables are actually used in the local function `computeRankMatrix`, so they have to be declared in the scope. It is extremely hard to notice that by reading the code, especially since `computeRankMatrix` is called in the block above the one where they are declared. I guess in Python, you can declare a variable in an `if` and use it outside, but not 2 `if`s deep.
Anyways, it's a real anti-pattern that should be reworked at some point.